src/azure-cli/azure/cli/command_modules/acr/token.py

```diff
@@ -39,6 +50,53 @@ def acr_token_create(cmd,
         )
     )
 
+    if no_passwords:
+        return poller
+
+    token = LongRunningOperation(cmd.cli_ctx)(poller)
+    _ensure_default_passwords(cmd, resource_group_name, registry_name, token, logger)
+    return token
+
+
+def _ensure_default_scope_map(cmd, resource_group_name, registry_name, token_name, repositories, logger):
+    import re
+    from ._client_factory import cf_acr_scope_maps
+    scope_map_name = DEF_SCOPE_MAP_NAME_TEMPLATE.format(token_name)
+    scope_map_name = re.sub(r'[^a-zA-Z0-9-]', '-', scope_map_name)
```

Comment:
`scope_map_name` is derived from `<tokenName>-scope-map`.
If there are non permissible characters found in the `<tokenName>`, the token creation itself will fail.

Therefore, it might lead to a case where a scope map gets created but the token creation fails.

@ankurkhemani any thoughts?

Reply:
Do token and scope-name share the same character set?

Reply:
Both tokens and scope-,map are ARM resources. Hence, the character set constraints for Scope map and Token is same as the constraints for any other ARM resource.

Reply:
Python SDK already has a validation for Token and Scope map resources (both are ARM resources and thus, there's also and existing ARM resource validation in RP).
In my opinion, we shouldn't perform any client side modifications such as    `scope_map_name = re.sub(r'[^a-zA-Z0-9-]', '-', scope_map_name)`.

If creation of `<tokenName>-scope-map` fails, then the operation should be stopped at that point with a suitable error message. 

@ankurkhemani please suggest changes here if needed.

Reply:
Agreed. I will pull out of the client side modification.